### PR TITLE
Pass tags on run creation

### PR DESCRIPTION
**Pull request by mociarain**

### CODE REVIEW

**Comment by mociarain** [comment]
Needs https://github.com/facultyai/faculty/pull/81 merged to master for tests to pass

File: mlflow_faculty/trackingstore.py [hunk 1]
```diff
@@ -198,14 +198,18 @@ def create_run(
         :param experiment_id: ID of the experiment for this run
         :param user_id: ID of the user launching this run
         :param source_type: Enum (integer) describing the source of the run
+        :param tags: List of Mlflow Tag entities.
+
```
Comment (acroz) [inline mlflow_faculty/trackingstore.py]: Remove extra empty line here?

File: tests/fixtures.py [hunk 2]
```diff
@@ -1,14 +1,38 @@
 from datetime import datetime
+from pytz import UTC
+from uuid import uuid4
 
-from mlflow.entities import Metric, Param, RunTag
+from mlflow.entities import (
+    Experiment as MLExperiment,
```
Comment (acroz) [inline tests/fixtures.py]: Elsewhere in this module we rename the Faculty classes and keep the MLflow classes at their original names. Can we do the same here?

File: tests/fixtures.py [hunk 3]
```diff
@@ -1,14 +1,38 @@
 from datetime import datetime
+from pytz import UTC
```
Comment (acroz) [inline tests/fixtures.py]: `pytz` is a third party library - can we put it in the same section as `faculty` and `mlflow`?

File: mlflow_faculty/mlflow_converters.py [hunk 4]
```diff
@@ -77,27 +78,35 @@ def faculty_run_to_mlflow_run(faculty_run):
...
 
-def mlflow_run_metric_to_faculty_run_metric(mlflow_run_metric):
-    return FacultyMetric(
-        key=mlflow_run_metric.key,
-        value=mlflow_run_metric.value,
-        timestamp=mlflow_timestamp_to_datetime_seconds(
-            mlflow_run_metric.timestamp
-        ),
-    )
+def mlflow_metrics_to_faculty_metrics(mlflow_metrics):
```
Comment (acroz) [inline mlflow_faculty/mlflow_converters.py]: Why did you change these to handle lists instead of single objects? AFAICT all the other converters convert single objects, which seems like a clearer API.

File: tests/fixtures.py [hunk 5]
```diff
@@ -1,14 +1,38 @@
...
 )
+from mlflow_faculty.py23 import to_timestamp
 
-from pytz import UTC
 
+PROJECT_ID = uuid4()
+STORE_URI = "faculty:{}".format(PROJECT_ID)
+
+EXPERIMENT_ID = 12
+CREATED_AT = datetime.now(tz=UTC)
```
Comment (acroz) [inline tests/fixtures.py]: AFAICT this (and LAST_CREATED_AT below) are not actually used. Can we remove?

File: mlflow_faculty/trackingstore.py (hunk 1 above)
Comment (mociarain) [reply mlflow_faculty/trackingstore.py]: 👍

File: tests/fixtures.py (hunk 2 above)
Comment (mociarain) [reply tests/fixtures.py]: 👍

File: tests/fixtures.py (hunk 3 above)
Comment (mociarain) [reply tests/fixtures.py]: 👍

File: mlflow_faculty/mlflow_converters.py (hunk 4 above)
Comment (mociarain) [reply mlflow_faculty/mlflow_converters.py]: I realised that we always want to convert tags never a single tag. The idea of single tag is kind of meaningless.

Same applies to Metrics and Params

File: tests/fixtures.py (hunk 5 above)
Comment (mociarain) [reply tests/fixtures.py]: 👍

File: mlflow_faculty/mlflow_converters.py (hunk 4 above)
Comment (pbugnion) [reply mlflow_faculty/mlflow_converters.py]: > The idea of single tag is kind of meaningless

Not sure that's true.

In general, it seems more versatile to have the converters act on individual objects, not collections.

Comment (mociarain) [reply mlflow_faculty/mlflow_converters.py]: We could leave it as it and then expose the single methods as and when they're needed?

Comment (acroz) [reply mlflow_faculty/mlflow_converters.py]: Pascal and I agreed to revert to the single-entity converters - can discuss why when you're back in the office. For now I've picked up this PR so we can push it through.

**Comment by pbugnion** [comment]
This looks good apriori. Did you test it against atlas at all?

**Comment by acroz** [comment]
> This looks good apriori. Did you test it against atlas at all?

Not yet, but will do after lunch.

**Comment by acroz** [comment]
@pbugnion I've tested that this works - can you approve if happy?
